addons/im_livechat/tests/test_chatbot_internals.py

```diff
@@ -108,3 +109,74 @@ def test_chatbot_welcome_steps(self):
         welcome_steps = self.chatbot_script._get_welcome_steps()
         self.assertEqual(len(welcome_steps), 1)
         self.assertEqual(welcome_steps, self.chatbot_script.script_step_ids[0])
+
+    def test_chatbot_multiple_rules_on_same_url(self):
+        bob_user = new_test_user(
+            self.env, login="bob_user", groups="im_livechat.im_livechat_group_user,base.group_user"
+        )
+        chatbot_no_operator, chatbot_operator = tuple(
+            self.env["chatbot.script"].create(
+                [
+                    {
+                        "title": "Chatbot operators not available",
+                        "script_step_ids": [
+                            Command.create(
+                                {
+                                    "step_type": "text",
+                                    "message": "I'm shown because there is no operator available",
+                                }
+                            )
+                        ],
+                    },
+                    {
+                        "title": "Chatbot operators available",
+                        "script_step_ids": [
+                            Command.create(
+                                {
+                                    "step_type": "text",
+                                    "message": "I'm shown because there is an operator available",
+                                }
+                            )
+                        ],
+                    },
+                ]
+            )
+        )
```

Comment:
Nitpick: single assignment makes the code harder to read. Would be simpler with 2 assignments:

```py
chatbot_no_operator = self.env["chatbot.script"].create()
chatbot_operator = self.env["chatbot.script"].create()
```

Less indent level, easier to spot data of the given record.